pkg/kubectl/cmd/wait/wait.go

```diff
@@ -345,6 +346,27 @@ func (w ConditionalWait) IsConditionMet(info *resource.Info, o *WaitOptions) (ru
 		cancel()
 		switch {
 		case err == nil:
+			if watchEvent.Type == watch.Error {
+				// the object should be status
+				if err := apierrors.FromObject(watchEvent.Object); err != nil {
+					return gottenObj, false, err
+				}
+				// TODO: this is terrible because I should be getting metav1.Status from the watch scheme - we could
```

Comment:
I think I'm going to move this into apierrors.FromObject, as much as it pains me, because that's at least guaranteed of having a chance to catch it.  I don't like hardcoding apiVersion, but that's the api...